Server/JS/db/NQLParser.js

```diff
@@ -0,0 +1,308 @@
+        return value;
+    }
+
+    /**
+     * Helper function that parse the side of a simple binary expression.
+     * @returns the parsed side.
+     */
+    _parse_side() {
+        let token = this._token_stream.next();
+        if (token.type == "identifier") {
+            return this._parse_identifier(token.value);
+        } else if (token.type == "int" || token.type == "float" || token.type == "string") {
+            return token;
+        }
+        this._throw(token, "an identifier, a number or a string");
+    }
+
+    _parse_operator_expression() {
+        let value = {
+            type    : NQLParser.TYPES.OPERATOR_EXPRESSION,
+            left    : this._parse_side(),
+            operator: this._token_stream.next()
+        };
+
+        if (!value.operator) {
+            this._throw(null, "an operator");
+        }
+
+        // TODO : support IN
+        if (value.operator.type == "operator") {
+            value.right = this._parse_side();
+            value.operator = value.operator.value;
+            return value;
+        } else {
+            this._throw(value.operator, "an operator");
+        }
+    }
+
+    _parse_binary_expression() {
+        let value;
+
+        // TODO : support NOT
+        if (this._token_stream.peek().type == NQLParser.TYPES.PARENTHESIS) {
+            value = this._parse_parenthesis();
+        } else {
+            value = this._parse_operator_expression();
+        }
+
+        if (this._token_stream.eof()) {
+            return value;
+        }
+
+        if (this._token_stream.peek().type == "operator") {
+            this._throw(this._token_stream.next());
+        }
+        if (this._token_stream.peek().type == "keyword") {
+            if (this._token_stream.peek().value.toUpperCase() == "AND" || this._token_stream.peek().value.toUpperCase() == "OR") {
+                value = {
+                    type    : NQLParser.TYPES.BINARY_EXPRESSION,
+                    left    : value,
+                    operator: this._token_stream.next().value.toUpperCase(),
+                    right   : this._parse_binary_expression()
+                };
+
+                // The aims of those following lines is to ensure that the operators will be interprated in the right order.
+                value.precedence = NQLParser.PRECEDENCE[value.operator];
+                if (value.right.precedence && value.precedence > value.right.precedence) {
+                    value.right.left = {
+                        type    : NQLParser.TYPES.BINARY_EXPRESSION,
+                        left    : value.left,
+                        operator: value.operator,
+                        right   : value.right.left
+                    };
+                    value = value.right;
+                }
+            }
+        }
+        return value;
+    }
+
+    //=====// Generic helpers //=====//
+
+    /**
+     * Helper function that build an error message an then throw an {@link UnexpectedTokenError}

```

Comment:
an error message an then  → an error message and then

Reply:
fixed